core/Azure.Mcp.Core/src/Areas/Server/Commands/ToolLoading/ServerToolLoader.cs

```diff
@@ -245,7 +253,8 @@ private async Task<CallToolResult> InvokeChildToolAsync(RequestContext<CallToolR
             }
 
             // At this point we should always have a valid command (child tool) call to invoke.
-            Activity.Current?.SetTag(TagName.IsServerCommandInvoked, true);
+            Activity.Current?.SetTag(TagName.IsServerCommandInvoked, true)?.SetTag(TagName.ToolName, command);
```

Comment:
You don't need the second null-conditional:

```suggestion
            Activity.Current?
                .SetTag(TagName.IsServerCommandInvoked, true)
                .SetTag(TagName.ToolName, command);
```